import-automation/progress-dashboard-rest/test/test_progress_log_database.py

```diff
@@ -0,0 +1,103 @@
+# limitations under the License.
+
+"""
+Tests for progress_log_database.py.
+"""
+
+import unittest
+from unittest import mock
+
+from google.cloud import exceptions
+
+from test import utils
+from app.model import progress_log_model
+from app.service import progress_log_database
+
+
+_MODEL = progress_log_model.ProgressLogModel
+
+
+class ProgressLogDatabaseTest(unittest.TestCase):
+    """Tests for BaseDatabase."""
+
+    @classmethod
+    def setUpClass(cls):
+        cls.emulator = utils.start_emulator()
+
+    @classmethod
+    def tearDownClass(cls):
+        utils.terminate_emulator(cls.emulator)
+
+    @mock.patch('app.utils.create_datastore_client',
+                utils.create_test_datastore_client)
+    @mock.patch('app.service.log_message_manager.LogMessageManager',
+                utils.LogMessageManagerMock)
+    def setUp(self):
+        """Ingests several logs before every test."""
+        self.database = progress_log_database.ProgressLogDatabase()
+        logs = [self.database.get(make_new=True) for _ in range(4)]
+        logs[0].update({_MODEL.level: 'info', _MODEL.message: 'first'})
+        logs[1].update({_MODEL.level: 'warning', _MODEL.message: 'second'})
+        logs[2].update({_MODEL.level: 'warning', _MODEL.message: 'third'})
+        logs[3].update({_MODEL.level: 'severe', _MODEL.message: 'fourth'})
+        self.logs_save_content = [logs[0], logs[1]]
+        self.logs_not_save_content = [logs[2], logs[3]]
+        for i, log in enumerate(self.logs_save_content):
+            self.logs_save_content[i] = self.database.save(
+                log, save_content=True)
+        for i, log in enumerate(self.logs_not_save_content):
+            self.logs_not_save_content[i] = self.database.save(
+                log, save_content=False)
+
+    def test_load_content(self):
+        """Tests that get with load_content=True loads the message."""
+        expected = ['first', 'second']
+        for i, message in enumerate(expected):
+            log = self.logs_save_content[i]
+            retrieved = self.database.get(
+                entity_id=log[_MODEL.log_id], load_content=True)
+            self.assertEqual(message, retrieved[_MODEL.message])
+
+    def test_not_load_content(self):
+        """Tests that get with load_content=False does not load the message."""
+        for log in self.logs_save_content:
+            log_id = log[_MODEL.log_id]
+            retrieved = self.database.get(entity_id=log_id, load_content=False)
+            self.assertEqual(log_id, retrieved[_MODEL.message])
+
+    def test_not_save_content(self):
+        """Tests that save with save_content=False does not save the message to
+        a bucket."""
+        expected = ['third', 'fourth']
+        for i, message in enumerate(expected):
+            log_id = self.logs_not_save_content[i][_MODEL.log_id]
+            retrieved = self.database.get(entity_id=log_id, load_content=False)
```

Comment:
shouldn't this case have load content = true to show that the save really didnt save because the load found nothing?

Reply:
Yes the next line is a test that asserts if load_content=True, an exception will be thrown.